### ISSUE TITLE
Generic Methods can't implement non-generic methods

### ISSUE THREAD

**Issue by VBAndCs** [dotnet/roslyn#59686]
Given these two interfaces:
```
Interface IFoo1
    Sub Test(x As Integer)
End Interface


Interface IFoo2(Of T)
    Sub Test(x As T)
End Interface
```

And this implementation:
```
Class Foo
    Implements IFoo1, IFoo2(Of Integer)

    Public Sub Test(Of T)(x As T) Implements IFoo1.Test, IFoo2(Of Integer).Test
   
    End Sub

End Class
```

Wht can't the generic metjhod `Test(Of T)` implement both interfaces?
It berfictly provides the minimum rewuirements of them, as It can be called with an integer without even specifieng the type param:
```
Dim f as New Foo
f.Test(5)
``

**Comment by VBAndCs** [comment]
Also, a Function should implement a Sub with the same param list, as the caller can ignore the return value. The Interface requirement is fulfiled.

**Comment by CyrusNajmabadi** [comment]
> Also, a Function should implement a Sub with the same param list, as the caller can ignore the return value. The Interface requirement is fulfiled.

This is the same issue. The signatures must match. 

You could ask for a language change here. The lang could say that it emits two methods. One which implements the interface method, and one which doesn't. The first would then call the second, ignoring it's result. 

This would be a language change though, so you'd need to file it over at dotnet/vblang. Thanks!

**Comment by VBAndCs** [comment]
@CyrusNajmabadi 
> The runtime requires this.

As far as I know, there is no type params at runtime, as the compiler crates versions of the method of the assigned type param. So, There will be a `Test(Of Integer)(x As Integer) ` at runtime. If this is true, then it is perfect tp implement the interface.

**Comment by VBAndCs** [comment]
> You could ask for a language change here. The lang could say that it emits two methods. One which implements the interface method, and one which doesn't. The first would then call the second, ignoring it's result.

I'd prefer to have a relaxed Interfaces that accept any suitable sugnature, incuding widening conversions of params.
Anyway, your idea seeams nice. I am allowing my VB Record generator syntax to implement interfaces just by matching name and signatures of properties and methods. Something like this:
```VB
Record Test(
    Name$,
    Pos As (X%, Y%),
    Age%,
    Move = Sub(x%, y%) _Pos = (x, y),
    Move = Sub(x%) _Pos.X = x,
    GetText = Function( ) "Hello",
    Concat = Function(x$, y$) x & y,
) Inherits TestBase, Implements ITest(Of Integer)
```
I can make many relaxations, but vb compiler will give errors on the generated code, so, I can generate extar methods to fit the interface signature, and make them private (VB allows it). These private methods will just call the original methods if they are invoked via the interface. I like it.
Thanks..

**Comment by VBAndCs** [comment]
Done
https://www.nuget.org/packages/Visual-Basic-Record-Generator/2.0.0
VB record generator now allows readonly properties to implement read-write properties, and allows Functions to implement Subs.

**Comment by CyrusNajmabadi** [comment]
> If this is true, 

This is not true. You are confusing a particular instantiation if a method with the method itself. The runtime does not support this as I said.

**Comment by CyrusNajmabadi** [comment]
> I'd prefer to have a relaxed Interfaces that accept any suitable sugnature, incuding widening conversions of params.

That's fine. But that will have to go through a language change request.
